dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/IdentifiableObjectBundleHook.java

```diff
@@ -73,6 +78,58 @@ public void preCreate(IdentifiableObject identifiableObject, ObjectBundle bundle
     handleAttributeValues(identifiableObject, bundle, schema);
     handleSkipSharing(identifiableObject, bundle);
     handleSkipTranslation(identifiableObject, bundle);
+    handleSortOrder(identifiableObject, bundle, schema);
+  }
+
+  /**
+   * This method loops through all sortable List of given object and sets the sortOrder value for
+   * each item in the List if it is not already set.
+   *
+   * @param identifiableObject Object to set sortOrder on
+   * @param bundle {@link ObjectBundle}
+   * @param schema Schema of given object
+   */
+  private void handleSortOrder(
```

Comment:
Seeing as `BaseNameableObject` now implements `SortableObject`, does that now mean that some objects will now have a sort order whether they need/want it? (I'm guessing that not all `BaseNameableObject`s need sort order). Just trying to understand the bigger picture for this.
So some objects will have a sortOder:
- which is used in the UI
- which is purely used for use in the DB?

Reply:
I didn't make `BaseNameableObject` to implement `SortableObject`, only `Option` implement it. 

```
public class Option extends BaseNameableObject implements MetadataObject, SortableObject {
```

Reply:
sorry, missed that :)